todos-app/src/App/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,30 @@
+//import { Todo, AppState } from "../../DataStructure";
+import React, { ReactElement } from "react";
+import { ITodo, TodoListType } from "../../DataStructure";
+import TodoListItem from "./TodoListItem/TodoListItem";
+
+const TodoList: React.FC<TodoListType> = ({ todos }) => {
+  function toggleAllCheckbox(e: React.ChangeEvent<HTMLInputElement>): void {
+    //TODO write function
+  }
+
+  return (
+    <section className="main">
+      <input
+        id="toggle-all"
+        className="toggle-all"
+        type="checkbox"
+        onChange={toggleAllCheckbox}
+      />
+      <label htmlFor="toggle-all">Mark all as complete</label>
+      <ul className="todo-list">
+        {todos.map((todo: ITodo): ReactElement => {
```

Comment:
don't need types and there is a shorthand when a single value is returned. todos.map((todo) => <TodoListItem ...>); Should help with cleaning up unnecessary code.

Reply:
Fixed it